sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,744 @@
+char telemetry_topic[128];
+static const az_span telemetry_name = AZ_SPAN_LITERAL_FROM_STR("temperature");
+static char telemetry_payload[32];
+
+// IoT Hub Direct Methods Values
+static char methods_response_topic[128];
+static const az_span report_method_payload_value_span = AZ_SPAN_LITERAL_FROM_STR("since");
+static const az_span report_method_name_span = AZ_SPAN_LITERAL_FROM_STR("getMaxMinReport");
+static const az_span report_max_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("maxTemp");
+static const az_span report_min_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("minTemp");
+static const az_span report_avg_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("avgTemp");
+static const az_span report_start_time_name_span = AZ_SPAN_LITERAL_FROM_STR("startTime");
+static const az_span report_end_time_name_span = AZ_SPAN_LITERAL_FROM_STR("endTime");
+static char end_time_buffer[32];
+static char methods_response_payload[256];
+
+// IoT Hub Twin Values
+static char reported_property_topic[128];
+static const az_span desired_temp_property_name = AZ_SPAN_LITERAL_FROM_STR("targetTemperature");
+static const az_span max_temp = AZ_SPAN_LITERAL_FROM_STR("maxTempSinceLastReboot");
+static az_span reported_property_topic_request_id = AZ_SPAN_LITERAL_FROM_STR("reported_prop");
+static char reported_property_payload[32];
+
+// PnP Device Values
+static int32_t current_device_temp = DEAFULT_START_TEMP_CELSIUS;
+static int32_t device_temperature_avg_total = DEAFULT_START_TEMP_CELSIUS;
+static uint32_t device_temperature_avg_count = 1;
+static int32_t device_max_temp = DEAFULT_START_TEMP_CELSIUS;
+static int32_t device_min_temp = DEAFULT_START_TEMP_CELSIUS;
+static int32_t device_avg_temp = DEAFULT_START_TEMP_CELSIUS;
+
+//
+// Configuration and connection functions
+//
+static az_result read_configuration_and_init_client();
+static az_result read_configuration_entry(
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span buffer,
+    az_span* out_value);
+static az_result create_mqtt_endpoint(char* destination, int32_t destination_size, az_span iot_hub);
+static int connect_device();
+static int subscribe();
+
+//
+// Messaging functions
+//
+static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message);
+static int send_telemetry_messages();
+static int send_method_response(
+    az_iot_hub_client_method_request* request,
+    uint16_t status,
+    az_span response);
+static int send_reported_temperature_property(double desired_temp);
+static az_result parse_desired_temperature_property(az_span twin_span, double* parsed_value);
+static az_result report_method(az_span payload, az_span response, az_span* out_response);
+
+int main()
+{
+  int rc;
+
+  // Get the boot time for methods response
+  time_t rawtime;
+  struct tm* timeinfo;
+  time(&rawtime);
+  timeinfo = localtime(&rawtime);
+  size_t len = strftime(boot_time_str, sizeof(boot_time_str), "%Y-%m-%dT%H:%M:%S%z", timeinfo);
```

Comment:
Nit - shouldn't `%Y-%m-%dT%H:%M:%S%z` itself be a constant defined outside of main() itself?

Reply:
great call yes it should